src/resources/Fields/Fields.ts

```diff
@@ -0,0 +1,47 @@
+import API from '../../APICore';
+import {PageModel} from '../BaseInterfaces';
+import Indexes from '../Indexes/Indexes';
+import Resource from '../Resource';
+import {FieldModel, ListFieldsParams} from './FieldsInterfaces';
+
+export default class Field extends Resource {
+    static baseUrl = `/rest/organizations/${API.orgPlaceholder}/indexes/fields`;
+
+    create(field: FieldModel) {
+        return this.api.post(Field.baseUrl, field);
+    }
+
+    createFields(fields: FieldModel[]) {
+        return this.api.post(`${Field.baseUrl}/batch/create`, fields);
+    }
+
+    deleteFields(fieldIds: string[]) {
+        const params = fieldIds.filter(Boolean).toString();
+
+        return this.api.delete(`${Field.baseUrl}/batch/delete?fields=${new URLSearchParams(params)}`);

```

Comment:
It would be cool if you could integrate that logic under the `buildPath` method. This way we could reuse it in other calls.